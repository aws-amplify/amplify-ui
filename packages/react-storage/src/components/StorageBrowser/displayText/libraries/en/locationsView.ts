import { DEFAULT_LIST_VIEW_DISPLAY_TEXT } from './shared';
import { DefaultLocationsViewDisplayText } from '../../types';
import { Permission } from '../../../storage-internal';

const PERMISSION_DISPLAY_TEXT: Record<Permission, string> = {
  READ: 'Read',
  WRITE: 'Write',
  READWRITE: 'Read/Write',
};

export const DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT: DefaultLocationsViewDisplayText =
  {
    ...DEFAULT_LIST_VIEW_DISPLAY_TEXT,
    title: 'Home',
    searchPlaceholder: 'Filter folders and files',
    getListResultsMessage: () => 'lol',
    getPermissionName: (permission: Permission) =>
      PERMISSION_DISPLAY_TEXT[permission] ?? permission,
    getDownloadLabel: (fileName: string) => `Download ${fileName}`,
    tableColumnBucketHeader: 'Bucket',
    tableColumnFolderHeader: 'Folder',
    tableColumnPermissionsHeader: 'Permissions',
    tableColumnActionsHeader: 'Actions',
  };
