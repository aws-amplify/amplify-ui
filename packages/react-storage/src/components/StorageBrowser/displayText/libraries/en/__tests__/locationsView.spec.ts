import { LIST_LOCATIONS_SCENARIOS } from './scenarios';
import { DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT } from '../locationsView';

describe('LocationsView display text', () => {
  it('should match snapshot values', () => {
    expect(DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT).toMatchSnapshot();
  });

  it.each(LIST_LOCATIONS_SCENARIOS)(
    '`getListLocationsResultMessage` returns the expected values in the %s scenario',
    (_, data) => {
      const { getListLocationsResultMessage } =
        DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT;

      expect(getListLocationsResultMessage(data)).toMatchSnapshot();
    }
  );

  it('returns the expected values from getPermissionName', () => {
    const { getPermissionName } = DEFAULT_LOCATIONS_VIEW_DISPLAY_TEXT;

    expect(getPermissionName(['get', 'list'])).toMatchSnapshot();
    expect(getPermissionName(['write', 'delete'])).toMatchSnapshot();
    expect(
      getPermissionName(['delete', 'get', 'list', 'write'])
    ).toMatchSnapshot();
  });
});
