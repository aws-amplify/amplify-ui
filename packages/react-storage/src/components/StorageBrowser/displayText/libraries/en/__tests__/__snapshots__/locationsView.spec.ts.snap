// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsView display text \`getListLocationsResultMessage\` returns the expected values in the empty results scenario 1`] = `
{
  "content": "No folders or files.",
  "type": "info",
}
`;

exports[`LocationsView display text \`getListLocationsResultMessage\` returns the expected values in the empty search results scenario 1`] = `
{
  "content": "No folders or files.",
  "type": "info",
}
`;

exports[`LocationsView display text \`getListLocationsResultMessage\` returns the expected values in the failed scenario 1`] = `
{
  "content": "Network got confused",
  "type": "error",
}
`;

exports[`LocationsView display text \`getListLocationsResultMessage\` returns the expected values in the loading scenario 1`] = `undefined`;

exports[`LocationsView display text \`getListLocationsResultMessage\` returns the expected values in the search failed scenario 1`] = `
{
  "content": "Network got confused",
  "type": "error",
}
`;

exports[`LocationsView display text \`getListLocationsResultMessage\` returns the expected values in the search limit exhausted scenario 1`] = `
{
  "content": "No more folders or files found by something to look for.",
  "type": "warning",
}
`;

exports[`LocationsView display text returns the expected values from getPermissionName 1`] = `"Read"`;

exports[`LocationsView display text returns the expected values from getPermissionName 2`] = `"Write"`;

exports[`LocationsView display text returns the expected values from getPermissionName 3`] = `"Read/Write"`;

exports[`LocationsView display text should match snapshot values 1`] = `
{
  "getDownloadLabel": [Function],
  "getListLocationsResultMessage": [Function],
  "getPermissionName": [Function],
  "loadingIndicatorLabel": "Loading",
  "searchClearLabel": "Clear search",
  "searchPlaceholder": "Filter folders and files",
  "searchSubmitLabel": "Submit",
  "tableColumnActionsHeader": "Actions",
  "tableColumnBucketHeader": "Bucket",
  "tableColumnFolderHeader": "Folder",
  "tableColumnPermissionsHeader": "Permissions",
  "title": "Home",
}
`;
